tests/resources/valid/class/generics_check.py

```diff
@@ -1,52 +1,79 @@
-# This class has no state
+from collections import defaultdict
+from typing import Callable
+from typing import Optional
+
+
+class Err1(Exception):
+    def __init__(self, msg: str):
+        super().__init__(msg)
+
+
+class Err2(Exception):
+    def __init__(self, msg: str):
+        super().__init__(msg)
+
+
+class MyType:
+    super_field = None
+
+    def __init__(self, super_field: str):
+        self.super_field = super_field
+
+
 class MyClass2(MyType):
-    _z_modified = None
-    _other_field = 10
+    _z_modified: str = f"asdf"
+    _other_field: int = 10
+
+    def __init__(self, other_field: int, z: int):
+        super().__init__(f"the quick brown fox jumped over the slow donkey")
+        if z > 10:
+            raise Err1(f"Something is wrong!")
+        self.z_modified = f"fdsa"
 
-    def __init__(self, other_field, z):
-        super().__init__(self)
-        if z > 10: raise Err("Something is wrong!")
-        self.z_modified = z * SOME_CONSTANT
+        (a, b) = (10, 20)
+        (a, b) = (30, 40)
+        (a, b) = (0, 10)
 
-        something = {
-            b : d, other : a
-        }[c]
+        something = defaultdict(lambda: z + 100, {
+            10: z + 10,
+        })[z]
+
+        my_bool = True
+        other = {
+            True: 2,
+            False: 3,
+        }[my_bool]
 
         a = None
         try:
-            a = self.z_modified
-        except Exception as err1:
-            print("hey")
-            print("there")
-        except MyErr as err2:
-            print("hoi")
+            a = self.error_function()
+        except Err1 as err1:
+            print(err1)
+            a = -1
+        except Err2 as err2:
+            print(err2)
+            a = -2
+
+        print(a)
+
+    def error_function(self) -> int: return 200
 
-    def connect(self):
-        self.other_field = 200
+    def connect(self): self.other_field = 200
 
-    def fun_a(self):
-        print(self)
+    def fun_a(self): print(self)
 
-    def _fun_b(self):
-        print("this function is private!")
+    def _fun_b(self): print(f"this function is private!")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/app/JSAbrahams/mamba/pullRequest?prid=4098440)